modules/data_integrity_flag/jsx/setFlagForm.js

```diff
@@ -0,0 +1,113 @@
+/* exported SetFlagForm */
+
+class SetFlagForm extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      formData: {}
+    };
+
+    // Bind component instance to custom methods
+    this.setFormData = this.setFormData.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  /**
+   * Set the form data based on state values of child elements/componenets
+   *
+   * @param {string} formElement - name of the selected element
+   * @param {string} value - selected value for corresponding form element
+   */
+  setFormData(formElement, value) {
+    const formData = this.state.formData;
+    formData[formElement] = value;
```

Comment:
In fact `React.addons` seems to be deprecated approach as well, so for now I just deep cloned the state as we do in most places.